### PR TITLE
Update to new aklogger structure

### PR DESCRIPTION
**Pull request by sks444**

### CODE REVIEW

File: aklogger/__init__.py [hunk 1]
```diff
@@ -1,57 +1,129 @@
...
+DEBUG = logging.DEBUG
+NOTSET = logging.NOTSET
+
+level = NOTSET
+slack_level = NOTSET
+slack_enable = False
+project_name = 'dummy'
+filename = 'dummy.log'
+file_handler = False
+formatter = '%(name)s - %(levelname)s - %(message)s'
```
Comment (cosmosgenius) [inline aklogger/__init__.py]: It's better to move these property to class level.
```
class AKLogger...:
	log_level = NOTSET
	...
```

File: aklogger/__init__.py [hunk 2]
```diff
@@ -1,57 +1,129 @@
...
+        self.logger = logging.getLogger(name)
+        if file_handler:
+            self.add_file_handler(filename)
 
-    def info(self, title=None, msg=None, *args, **kwargs):
-        message = utils.get_message(title, msg)
-        if not message:
-            return
-        return self.logger.info(message)
+    def debug(self, summary=None, details=None, channel='#error'):
```
Comment (cosmosgenius) [inline aklogger/__init__.py]: logging function needs to accept *args **kwargs.

File: aklogger/__init__.py [hunk 3]
```diff
@@ -1,57 +1,129 @@
...
+                attachments=[
+                    {
+                        'fallback': summary,
+                        'text': details,
+                        'color': color,
+                    }
+                ] if details else None,
+            )
+        except Exception as e:
+            self.logger.error(e)
```
Comment (cosmosgenius) [inline aklogger/__init__.py]: this should be formatted i think. Say 'Slack failed..' and then err obj

File: aklogger/__init__.py [hunk 4]
```diff
@@ -1,57 +1,129 @@
...
+
+
+def set_project(name):
```
Comment (cosmosgenius) [inline aklogger/__init__.py]: For these function create a global AKLogger instance and use that here.

File: aklogger/handlers.py [hunk 5]
```diff
@@ -1,141 +1,21 @@
...
-        trace = {
-            'fallback': fallback,
-            'text': text,
-            'color': COLORS.get(self.level, INFO_COLOR)
-        }
-        return trace
+    def __init__(self, project, name, slack_token=None,
+                 slack_level=logging.ERROR, file_name=None):
+        super().__init__()
+        aklogger.set_project(project)
```
Comment (cosmosgenius) [inline aklogger/handlers.py]: Here use the global instance. And lets make all the property optional and set the value if it exists.

**Comment by subho007** [comment]
LGTM @cosmosgenius 👍

**Comment by subho007** [comment]
Also @sks444 next time when you open a PR, make sure you have some description in the PR

File: aklogger/__init__.py [hunk 6]
```diff
@@ -1,57 +1,150 @@
...
-
-    def debug(self, title=None, msg=None, *args, **kwargs):
-        message = utils.get_message(title, msg)
-        if not message:
-            return
-        return self.logger.debug(message)
+    def __init__(self, name, parent=None):
+        self.set_name(name)
+        self.parent = parent
+        self.logger = get_logger(self.get_name())
```
Comment (sks444) [inline aklogger/__init__.py]: @cosmosgenius, there is bit of problem here, its always importing the celery logger when there is a celery installed(which will always be the case in mycroft). So in that case our normal logger is not logging to console until it's a celery task.

P.S. file log and slack push is working in both the cases.

**Pull request merged**